File: frontend/packages/loop-components/observation-components/src/features/trace-list/stores/trace.ts
// Copyright (c) 2025 coze-dev Authors
// SPDX-License-Identifier: Apache-2.0
import {
  type TraceContextType,
  useTraceContext,
} from '@/features/trace-list/contexts/trace-context';

export function useTraceStore(
  selector?: (state: TraceContextType) => TraceContextType,
) {
  const context = useTraceContext();
  if (!selector) {
    return context;
  }
  return selector(context);
}

File: backend/modules/prompt/pkg/consts/trace.go
// Copyright (c) 2025 coze-dev Authors
// SPDX-License-Identifier: Apache-2.0

package consts

const (
	SpanTypePromptExecutor = "prompt_executor"
	SpanTypeSequence       = "sequence"
)

const (
	SpanNamePromptExecutor = "PromptExecutor"
	SpanNamePromptHub      = "PromptHub"
	SpanNameSequence       = "Sequence"
	SpanNamePromptTemplate = "PromptTemplate"
)

const (
	SpanTagCallType             = "call_type"
	SpanTagDebugID              = "debug_id"
	SpanTagPromptVariables      = "prompt_variables"
	SpanTagMessages             = "messages"
	SpanTagPromptTemplate       = "prompt_template"
	SpanTagPromptID             = "prompt_id"
	SpanTagOverridePromptParams = "override_prompt_params"
)

const (
	SpanBaggagePromptKey     = "prompt_key"
	SpanBaggagePromptVersion = "prompt_version"
)

const (
	SpanTagCallTypePromptPlayground = "PromptPlayground"
	SpanTagCallTypePromptDebug      = "PromptDebug"
	SpanTagCallTypePTaaS            = "PTaaS"
	SpanTagCallTypeEvaluation       = "Evaluation"
)
